src/Security/MemberAuthenticator/MemberAuthenticator.php

```diff
@@ -62,26 +62,24 @@ public function authenticate($data, &$result = null)
     protected function authenticateMember($data, &$result = null, $member = null)
     {
         $email = !empty($data['Email']) ? $data['Email'] : null;
-        // Default success to false
-        $result = new ValidationResult();
+        $result = $result ?: ValidationResult::create();
 
         // Check default login (see Security::setDefaultAdmin())
-        $asDefaultAdmin = $email === DefaultAdminService::getDefaultAdminUsername();
+        $asDefaultAdmin = DefaultAdminService::isDefaultAdmin($email);
         if ($asDefaultAdmin) {
             // If logging is as default admin, ensure record is setup correctly
-            /** @var Member $member */
-            $service = Injector::inst()->get(DefaultAdminService::class);
-            $member = $service->findOrCreateDefaultAdmin();
-            $validAdmin = $service->validateDefaultAdmin($email, $data['Password']);
-            $result = $member->canLogIn();
-            //protect against failed login
-            if ($validAdmin->isValid() && $result->isValid()) {
-                return $member;
-            } else {
-                $result->addError(_t(
-                    'SilverStripe\\Security\\Member.ERRORWRONGCRED',
-                    "The provided details don't seem to be correct. Please try again."
-                ));
+            $member = DefaultAdminService::singleton()->findOrCreateDefaultAdmin();
+            $member->validateCanLogin($result);
```

Comment:
`$result = $member->validateCanLogin($result);`?
I'm not sure if this call does anything otherwise